File: 5.implementation/redsun/src/main/webapp/static/js/services/ClientsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('clientService', function($http, $rootScope, $window) {
	return {
	
		// Insert.
		addClient: function(client) {
			var urlServer = $rootScope.makePostURL("/client/insert");
			return $http.post(urlServer, client)
			.success(function(response, status, headers, config) {
				response = angular.fromJson(response);
				if(response && response.status > 0){
					$window.location.href =  $rootScope.makeGetURL("/client/listpage");
				} else {
					$window.alert(response.description);
				}
			})
			.error(function(response, status, headers, config) {
				$window.alert('Some issues occured, contact Admin for more details.');
			});
		}
		
		// Update.
		, updateClient: function($scope) {
			var urlServer = $rootScope.makePostURL("/client/update/" + $scope.client.id);
			return $http.put(urlServer, $scope.client)
			.success(function(response, status, headers, config) {
				response = angular.fromJson(response);
				if(response && response.status > 0){
					$window.location.href =  $rootScope.makeGetURL("/client/listpage");
				} else {
					$window.alert(response.description);
				}
			})
			.error(function(response, status, headers, config) {
				$window.alert('Some issues occured, contact Admin for more details.');
			});
		}
		
		// Delete.
		, deleteClient: function(clientId) {
			var urlServer = $rootScope.makePostURL("/client/delete/" + clientId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getByIdClient: function(clientId) {
			var url = $rootScope.makeGetURL("/client/getbyid/" + clientId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilterClient: function(itemsPerPage, pageNo, client) {
			var url = $rootScope.makeGetURL("/client/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: client });
		}
		
	}// return.
});

File: 5.implementation/redsun-service/src/main/java/com/redsun/service/service/ClientsService.java
package com.redsun.service.service;

import com.redsun.service.entities.Clients;
import com.redsun.service.entities.Result;

/**
 * Clients Service interface
 */
public interface ClientsService {
	
    // Insert.
	Result insert(final Clients clients) ;

    // Update.
	Result update(final Clients clients) ;

    // Delete.
	Result delete(final int id) ;

	// Get By Id.
	Result getById(int clientId, int id) ;

    // Exists.
	Result exists(final int id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listClientsForPageAndFilter(final int itemsPerPage, final int pageNo, final Clients clients);
	
	//list all
	Result listAll();

}
